components/code-editor/code-editor.component.ts

```diff
@@ -214,7 +214,9 @@ export class NzCodeEditorComponent implements OnDestroy, AfterViewInit {
     }
 
     if (this.nzFullControl && this.value) {
-      warn(`should not set value when you are using full control mode! It would result in ambiguous data flow!`);
+      if (typeof ngDevMode === 'undefined' || ngDevMode) {
```

Comment:
Extract a function here?

Reply:
Hey. That's not possible since functions are _runtime_ and `ngDevMode` is a compile time variable.

Terser doesn't call functions during minification & sealing to know their return type (e.g. if it's a static boolean value, like `function doSmth() { return someBoolean }`).

E.g. this is an example of how Material is doing that and they also do duplicate the same condition every time:

![image](https://user-images.githubusercontent.com/7337691/144209339-31f4465c-16be-4433-9cf9-10cd978e8473.png)

Reply:
So Terser would remove these `if` branches in prodution?

Reply:
Ok. I didn't know Terser would do that. Thanks for clarifying.